compiler/rustc_span/src/symbol.rs

```diff
@@ -2586,6 +2589,65 @@ impl fmt::Display for MacroRulesNormalizedIdent {
     }
 }
 
+/// An newtype around `Ident` that calls [Ident::normalize_to_macros_2_0] on
+/// construction for "item hygiene" comparisons.
+///
+/// Identifiers with same string value become same if they came from the same macro 2.0 macro
+/// (e.g., `macro` item, but not `macro_rules` item) and stay different if they came from
+/// different macro 2.0 macros.
+// FIXME: Use more often
+#[derive(Copy, Clone, Eq, PartialEq, Hash)]
+pub struct ModernIdent(pub Ident);
```

Comment:
```suggestion
pub struct Macros20NormalizedIdent(pub Ident);
```
The modern / legacy terminology for macros 2.0 / macro_rules is not used anymore.